pkg/controllers/dataexport/reconcile.go

```
@@ -627,7 +627,8 @@ func (c *Controller) getSnapshotDriverName(dataExport *kdmpapi.DataExport) (stri
 	if err != nil && !k8sErrors.IsNotFound(err) {
 		return "", nil
 	}
-	return "", fmt.Errorf("did not find any supported snapshot driver for snapshot storage class %s", dataExport.Spec.SnapshotStorageClass)
+	logrus.Warnf("did not find any supported snapshot driver for snapshot storage class %s", dataExport.Spec.SnapshotStorageClass)
```

The old logic of returning errors was fine right? Also in line#628 we need to return the error. With both these changes, there would be no need to check for an empty snapshot name in the caller.
Also in the caller now when snapshot name is empty (in some cases), this error msg `fmt.Errorf("failed to get snapshot driver name: %v", err)` will have an empty error and might be confusing to the user.

---

Fixed